### ISSUE TITLE
No diagnostic when completion fails because of a compiler error

### ISSUE THREAD

**Issue by adpi2** [scalacenter/scastie#689]
## Reproduction Step

1. Create a new snippet
2. Type this without "saving"

```scala
def foo(x1: Option[String]): Option[String] = ???

def foo(x2: String): String = ????

foo(1).


```

I don't get any auto-completion on `foo(1).`.

This is expected since `foo(1).` does not type-check but there is no clue explaining me why.

## Expectation

It would be great if we could have some diagnostics explaining why the completion failed.

**Comment by julienrf** [comment]
I agree that it would be good to have a way to signal to the user that the code did not compile (even better: report the compilation errors).
